### PR TITLE
fix for date out of range error

### PR DESCRIPTION
**Pull request by jonas7654**
I ran into an out of range error when gestern.day has a value of zero which happens if today.day is equal to 1.

To solve it I wrote if conditions to get the the last day of the month before and not a zero value.

### CODE REVIEW

File: IliasDownloaderUniMA/__init__.py [hunk 1]
```diff
@@ -137,7 +137,16 @@ def translate_date(self, datestr):
...
+
+		if today.day-1 == 0 & today.month-1 in [1,3,5,7,8,10,12]:
+			gestern = today.replace(day = 31).strftime("%d. %b %Y")
+		elif today.day-1 == 0 & today.month-1 not in [1,3,5,7,8,10,12]:
+			gestern = today.replace(day = 30).strftime("%d. %b %Y")
+		elif today.day-1 == 0 & today.month-1 == 2:
+			gestern = today.replace(day = 28).strftime("%d. %b %Y")
+		else:
+			gestern = today.replace(day = today.day-1).strftime("%d. %b %Y")
+
```
Comment (jhelgert) [inline IliasDownloaderUniMA/__init__.py:149]: Using datetimes's [timedelta](https://docs.python.org/3/library/datetime.html#datetime.timedelta)

``` python
gestern = (today - timedelta(1)).strftime("%d. %b %Y")
```
does the same and is much cleaner IMHO.

**Review by jhelgert — CHANGES_REQUESTED** [review]
Thanks for the pull request!

File: IliasDownloaderUniMA/__init__.py (hunk 1 above)
Comment (jhelgert) [inline IliasDownloaderUniMA/__init__.py:149]: Using datetimes's [timedelta](https://docs.python.org/3/library/datetime.html#datetime.timedelta)

```suggestion
	gestern = (today - timedelta(1)).strftime("%d. %b %Y")		
```

does the same and is much cleaner IMHO.

**Review by jhelgert — CHANGES_REQUESTED** [review]
Thanks for reporting the bug and your pull request! Would be happy to merge it with the use of `timedelta`.

File: IliasDownloaderUniMA/__init__.py (hunk 1 above)
Comment (jonas7654) [reply IliasDownloaderUniMA/__init__.py:149]: Thanks! I am quite new to Python and didn’t know about the timedelta function. It looks cleaner so I would also prefer your solution :-)
I quickly wrote a fix to get it to work on my machine.

**Comment by jonas7654** [comment]
Thanks!
Your solution is better. I am quite new to Python so I didn’t know about the timedelta function.

**Review by jhelgert — CHANGES_REQUESTED** [review]
Change
``` python
from datetime import datetime
```
to

``` python
from datetime import datetime, timedelta
```

and we're ready to merge :).

File: IliasDownloaderUniMA/__init__.py [hunk 2]
```diff
@@ -4,8 +4,8 @@
 from bs4 import BeautifulSoup
 from urllib.parse import urljoin
 from pathlib import Path as plPath
-from dateutil.parser import parse as parsedate
-from datetime import datetime
+from dateutil.parser import parse
```
Comment (jhelgert) [inline IliasDownloaderUniMA/__init__.py:7]: Should be

```suggestion
from dateutil.parser import parse as parsedate
```

(as before).

**Review by jhelgert — CHANGES_REQUESTED** [review]
One last change..

File: IliasDownloaderUniMA/__init__.py (hunk 2 above)
Comment (jonas7654) [reply IliasDownloaderUniMA/__init__.py:7]: Sorry! I don’t why know why it got removed.. it was not on purpose

**Pull request merged**